.gitlab-ci.yml

```
@@ -799,3 +799,93 @@ new-issue-on-failure:
       - develop
       - master
   dependencies: []
+
+
+# The following are windows runner setting
+
+# Gitlab windows virtual machine shared runner tag
+.shared_windows_runners:
+  tags:
+  - shared-windows
+  - windows
+  - windows-1809
+
+.windows_before_script_template: &windows_default_before_script
+  - choco install cmake -y
+  - $env:PATH="C:\Program Files\CMake\bin;$env:PATH"
+
+.windows_variables_template: &windows_default_variables
+  BUILD_TYPE: Debug
+  BUILD_SHARED_LIBS: "ON"
+  BUILD_CUDA: "OFF"
+
+.windows_ci_template: &windows_default_ci
+  stage: build
+  extends:
+    - .shared_windows_runners
+  # can not use rules and except together.
+  # use '$CI_PIPELINE_SOURCE == "schedule"' to do except: - schedule
+  # run pipeline automatically on master/develop but manually on other branches.
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "schedule"'
+      when: never
+    - if: '$CI_COMMIT_REF_NAME != "master" && $CI_COMMIT_REF_NAME != "develop"'
+      when: manual
+    - when: on_success
+  variables: *windows_default_variables
+  before_script: *windows_default_before_script
+  dependencies: []
+
+.windows_compile_template: &windows_default_compile
+  - mkdir -p ${CI_JOB_NAME} && cd ${CI_JOB_NAME}
+  - cmake ${CI_PROJECT_DIR}
+      -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}
+      -DGINKGO_DEVEL_TOOLS=OFF -DGINKGO_BUILD_REFERENCE=ON
+      -DGINKGO_BUILD_OMP=OFF -DGINKGO_BUILD_CUDA=${BUILD_CUDA}
+      -DGINKGO_BUILD_HIP=OFF
+      -DGINKGO_BUILD_TESTS=ON -DGINKGO_BUILD_EXAMPLES=ON
+  - cmake --build . --config ${BUILD_TYPE}
+
+# Windows runner does not contain nvidia device
+# Only compile cuda and reference.
+compile/cuda102/msvc2019/cuda/release/shared:
+  <<: *windows_default_ci
+  variables:
+    <<: *windows_default_variables
+    BUILD_CUDA: "ON"
+    BUILD_TYPE: Release
+  before_script:
+    - .windows_default_before_script
+    - curl.exe -L http://developer.download.nvidia.com/compute/cuda/10.2/Prod/local_installers/cuda_10.2.89_441.22_win10.exe --output cuda.exe
```

```suggestion
    - curl.exe -L https://developer.download.nvidia.com/compute/cuda/10.2/Prod/local_installers/cuda_10.2.89_441.22_win10.exe --output cuda.exe
```
for security reasons :)